### ISSUE TITLE
[UI] Add support for changing Cluster Template revision upon Cluster edit

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#21441]
It's not currently possible to edit a cluster in the UI and change the cluster template revision to something else and save. We would like to add this in, otherwise users won't be able to leverage new revisions of templates on existing clusters.

**Comment by davidnuzik** [comment]
Version: master-head (7/18/19)

Overrides do not change when I change up my templates. I think that the UI isn't refreshing and reading in the new template when switch them from the dropdown.

**Comment by davidnuzik** [comment]
Version: master-head (7/22/19) 

**Summary:**
This change resulted in adding an option to opt-in to picking a cluster template when you create or edit a cluster. An example is below. By default when creating a cluster this is unchecked.
![image](https://user-images.githubusercontent.com/45179589/61652918-e7177e00-ac6d-11e9-9faf-509c063ee275.png)

**Validation:**

_Case 1:_

I created a couple of revisions of a cluster template. One had k8s v1.14.x and the second had v1.15.x. I created the cluster with the 1.14 template (my first revision). I then edited the cluster and picked the second k8s 1.15 template (my second revision) as pictured below, this successfully changed the value of the k8s to 1.15.x:
![image](https://user-images.githubusercontent.com/45179589/61653046-3f4e8000-ac6e-11e9-9340-7aa5779b2bf3.png)

_Case 2 (failed, but not related to this issue):_

I created a template revision that sets various fields to custom values some with overrides and some without. I created a cluster with this template, while doing so I ensured all the settings took correctly and that overrides work correctly.
I then created a second revision that changes these fields again and inverts the fields I have set overrides for. I then edited the cluster and ensure all these settings took correctly and that overrides work correctly. -- This part failed. The http sent to PUT (edit) the cluster looks good, my answers and such are all there, but there is no actual changed observed in the UI. And the cluster is not updated per my new answers. Opened a new, separate issue for that here: https://github.com/rancher/rancher/issues/21680

For each test:
- Ensured answers appear correct (view template in API and review cluster create/edit http)
- Ensured POST/PUT for cluster creation/edit appears correct.
- Ensured each change to cluster (change template) successfully worked -- i.e. the requested change actually took effect.
- Ensured able to delete cluster without any issues.

Note: Due to bug https://github.com/rancher/rancher/issues/21679 Rancher UI will appear to not actually save cluster template revision when clone but it will actually create it.
